src/hotspot/share/code/codeCache.cpp

```diff
@@ -872,6 +872,12 @@ void CodeCache::on_gc_marking_cycle_finish() {
   update_cold_gc_count();
 }
 
+// Arm nmethods so that special actions are taken (nmethod_entry_barrier) for
+// on-stack nmethods. It's used in two places:
+// 1. Used before the start of concurrent marking so that oops inside on-stack
+//    nmethods are visited.
+// 2. Used at the end of (stw/concurrent) marking so that nmethod::_gc_epoch is
+//    up-to-date, which provides more accurate estimate of nmethod::is_cold.
```

Comment:
I would prefer to have general documentation and usage information about a method in the .hpp file.

Reply:
Moved.